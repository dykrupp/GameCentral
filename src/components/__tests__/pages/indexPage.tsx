import React from 'react';
import { render } from '@testing-library/react';
import * as Gatsby from 'gatsby';
import IndexPage from '../../../pages';
import { data } from '../__queryResults__/queryResult';

beforeEach(() => {
  const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');

  useStaticQuery.mockImplementationOnce(() => ({
    site: {
      siteMetadata: {
        title: 'GameCentral',
      },
    },
  }));

  useStaticQuery.mockImplementationOnce(() => data);

  useStaticQuery.mockImplementationOnce(() => ({
    site: {
      siteMetadata: {
        title: 'GameCentral',
      },
    },
  }));
});

describe('IndexPage', () => {
  it('renders', () => {
    const { asFragment } = render(<IndexPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays correct # of topPicks', () => {
    const { getByTestId } = render(<IndexPage />);
    const imageSliderEle = getByTestId('image-slider-container').children[0];
    const navEle = Array.from(imageSliderEle.children).filter(
      (x) => x.className === 'awssld__bullets'
    )[0];
    expect(navEle.children.length).toBe(5);
  });
});
