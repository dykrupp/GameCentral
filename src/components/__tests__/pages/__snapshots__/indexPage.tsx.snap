// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders 1`] = `
<DocumentFragment>
  <div
    class="sc-Axmtr bQvARb"
  >
    <h4
      class="MuiTypography-root sc-AxmLO gFZrrg MuiTypography-h4"
    >
      Our Top Picks
    </h4>
    <div
      class="MuiPaper-root sc-fzoLsD fXnotG MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        class="sc-AxjAm fdgoAF"
      >
        <div
          class="sc-AxirZ iWxEsR"
          data-testid="image-slider-container"
          style="display: block;"
        >
          <div
            class="awssld--organic-arrows awssld"
          >
            <div
              class="awssld__wrapper"
            >
              <div
                class="awssld__container"
              >
                <div
                  class="awssld__box awssld--active"
                >
                  <div
                    class="awssld__content"
                  >
                    <img
                      src="/static/a3cd88d54452b83e7ca10673b34bda29/32fd5/fl_test_jAqApqoVi12gf4bMCQwgKMEW.jpg"
                    />
                    <div
                      class="awssld__timer"
                      style="--timer-delay: 15000ms; --timer-height: 4px; --timer-background-color: #9A6DEA;"
                    />
                  </div>
                </div>
                <div
                  class="awssld__box"
                />
              </div>
              <div
                class="awssld__controls awssld__controls--hidden"
              >
                <button
                  aria-label="next"
                  class="awssld__next"
                >
                  <span
                    class="awssld__controls__arrow-right"
                  />
                </button>
                <button
                  aria-label="previous"
                  class="awssld__prev"
                >
                  <span
                    class="awssld__controls__arrow-left"
                  />
                </button>
              </div>
            </div>
            <nav
              class="awssld__bullets"
            >
              <button
                class="awssld__bullets--active"
                data-index="0"
              >
                0
              </button>
              <button
                data-index="1"
              >
                1
              </button>
              <button
                data-index="2"
              >
                2
              </button>
              <button
                data-index="3"
              >
                3
              </button>
              <button
                data-index="4"
              >
                4
              </button>
            </nav>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root makeStyles-accordion-1 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-summary-3"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-heading-2 MuiTypography-body1"
            >
              MORE INFORMATION
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="color: white;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-container MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root sc-AxhCb iBPfjP"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text sc-AxheI iUeGSr MuiSkeleton-pulse"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-fzozJi cqNSUU MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add To Cart
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
